manimlib/mobject/svg/text_mobject.py

```diff
@@ -66,8 +66,21 @@ def __init__(self, text, **config):
             self.set_color_by_t2g()
 
         # anti-aliasing
-        if self.height is None:
-            self.scale(TEXT_MOB_SCALE_FACTOR * self.font_size)
+        if self.height is None and not self.debug:
+            self.load_scale_factor()
+            self.scale(self.scale_factor * self.font_size)
+    
+    def load_scale_factor(self):
+        factor_file = os.path.join(get_temp_dir(), "text_scale_factor.txt")
+        if os.path.exists(factor_file):
+            with open(factor_file, 'r') as f:
+                self.scale_factor = float(f.read())
+        else:
+            tex_height = TexText("b", font_size=1).get_height()
```

Comment:
I worry this would mean that for rendering text LaTeX would be a must. Also, there should a better solution than this. Both the size needn't need to match but they should be made to fit into the screen.

Reply:
Since the size of `TexText` is the same on different machines, `tex_height` can be changed to a single number.
And if both the size matched, they are ensured to fit into the screen.

Reply:
I'm inclined to agree there's probably a better solution. Why is the right answer not to simply adjust what the constant is? Will the scale factor be different on different machines?

Reply:
> Will the scale factor be different on different machines?

No, I don't think so. I think adjusting the scale factor should do the trick.

Another possible solution I could think of is letting Pango handle font size in the SVG file(the `size` parameter there) and improving the SVG parser to consider the size or view box of the SVG file and not worry about scaling it. This would require a lot of work though.

Reply:
But in [ManimCommunity/ManimPango#52](https://github.com/ManimCommunity/ManimPango/issues/52), @GZTimeWalker proved that.
And on my old computer (Windows10, 1080p), the default size of `Text` is the same as `TexText`. On @GZTimeWalker 's computer, the default size of `Text` is four times larger than `TexText`. On my new computer (macOS 10.15, 3072x1920), the default size of `Text` is about two times larger than `TexText`.
So I think the scale factor will be different on different machines.

Reply:
> Another possible solution I could think of is letting Pango handle font size in the SVG file(the size parameter there) and improving the SVG parser to consider the size or view box of the SVG file and not worry about scaling it. This would require a lot of work though.

I think that's a good solution, that is to solve it in manimpango, which is also what @GZTimeWalker tends to.

Reply:
> And on my old computer (Windows10, 1080p), the default size of Text is the same as TexText. On @GZTimeWalker 's computer, the default size of Text is four times larger than TexText. On my new computer (macOS 10.15, 3072x1920), the default size of Text is about two times larger than TexText.

I don't think it is correct to compare `TexText` vs `Text`. Both can have different size but it needs to be scaled correctly when it is added to Manim. 
Also, I don't think Pango renders to a different size on different computers, I have tried then on Ubuntu and Windows, and get a very similar SVG file, so I think this is something to do with Manim's SVG parsing.
[svg.zip](https://github.com/3b1b/manim/files/6660556/svg.zip)

> ...that is to solve it in manimpango

That solution is to be implemented in `SVGMobject` and not in ManimPango. ManimPango already has the implementation for adjusting font size.

Reply:
I have something locally which may work. As in I didn't touch `SVGMobject`. Will make a PR soon so that you can test.

Reply:
@TonyCrane can you test https://github.com/3b1b/manim/pull/1553 to whether that works for you? At least, the text shouldn't go out of the frame.

Reply:
Sorry to see it so late. I just tested it and the results are very good. I think that is a good solution, so I plan to revert this pr, which will cause more problems.